housekeeper/store/crud/create.py

```diff
@@ -58,8 +58,8 @@ def add_bundle(self, data: dict) -> Tuple[Bundle, Version] | None:
     def _add_files_to_version(self, files: list[dict], version_obj: Version) -> None:
         """Create file objects and the tags and add them to a version object"""
 
-        tag_names = set(tag_name for file_data in files for tag_name in file_data["tags"])
-        tag_map = self._build_tags(tag_names)
+        tag_names = {tag_name for file_data in files for tag_name in file_data["tags"]}
+        tag_map = self._build_tags(list(tag_names))
```

Comment:
add the type hints

Reply:
Does the tag_names not count as using a constructor?

Reply:
Fixed